tests/repository_simulator.py

```diff
@@ -205,10 +205,15 @@ def _fetch(self, url: str) -> Iterator[bytes]:
         if path.startswith("/metadata/") and path.endswith(".json"):
             # figure out rolename and version
             ver_and_name = path[len("/metadata/") :][: -len(".json")]
+            # inside a version folder
+            if "/" in ver_and_name:
+                ver_and_name = ver_and_name.split("/")[1]
```

Comment:
What happens if there's multiple nested folders? `ver_and_name = "/foo/bar.json"`

I like 

```python
>>> "foo/bar/baz.json".partition("/")
('foo', '/', 'bar/baz.json')
>>> "foo/bar/baz.json".rpartition("/") 
('foo/bar', '/', 'baz.json')
```

For making intent a little clearer there.

You can also `assert len(ver_and_name.split("/")  <= 2`